Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM rust:1.69-bullseye as rust-build-stage
+RUN apt-get update && apt-get upgrade -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /build
+COPY Cargo.lock Cargo.toml ./
+COPY src src/
+COPY config config/
+COPY themes themes/
```

Comment:
this needs to be maintained in the future. is there any reason not to copy the full repo contents?

Reply:
`COPY . .` is dangerous in a single stage build. Since this is a multi-stage build, the risk is greatly reduced, but not eliminated.

1. If we don't have a `.dockerignore` file, the build context can grow over time. For instance, are you copying in the whole `.git/` directory to the build stage? Are you copying temporary files you have in the directory like large image files? All that busts the cache and the data gets sent to the docker engine. Will that matter? In this case, I'd say "probably not," buuuut if you're explicit about what you're copying in (and what you're not) then you're in control, now and in the future.
2. It's still conceivable that something bad will get baked into the run-stage by accident, like secrets or others. That rust compiles to a single binary is useful, but the point is that by being explicit you ensure the build process is deterministic and you are in control of what goes in and what doesn't.

Having said all that, for `vivid` none of this is a huge deal and I certainly won't die on this hill. If you are decidedly against this, I'm happy to just do `COPY . .` and call it a day.

Reply:
Thank you for the explanation — let's keep your version then. It would probably be a good idea to test the docker image build in CI to make sure it's maintained in the future? But no need to do this in this PR.